srtp/srtp.c

```diff
@@ -74,39 +74,50 @@ srtp_debug_module_t mod_srtp = {
 };
 
 #define octets_in_rtp_header 12
-#define uint32s_in_rtp_header 3
 #define octets_in_rtcp_header 8
-#define uint32s_in_rtcp_header 2
-#define octets_in_rtp_extn_hdr 4
+#define octets_in_rtp_xtn_hdr 4
+
+static uint32_t srtp_get_rtp_hdr_len(const srtp_hdr_t *hdr)
+{
+    return octets_in_rtp_header + 4 * hdr->cc;
+}
+
+static srtp_hdr_xtnd_t *srtp_get_rtp_xtn_hdr(srtp_hdr_t *hdr)
+{
+    uint32_t rtp_xtn_hdr_start = srtp_get_rtp_hdr_len(hdr);
+    return (srtp_hdr_xtnd_t *)((uint8_t *)hdr + rtp_xtn_hdr_start);
+}
+
+static uint32_t srtp_get_rtp_xtn_hdr_len(const srtp_hdr_xtnd_t *xtn_hdr)
```

Comment:
are you talking about srtp_get_rtp_xtn_hdr ?  then yes could return NULL,  if you are talking about srtp_get_rtp_xtn_hdr_len then there is no x bit here, it is assumed that the caller has checked the x bit in the rtp_hdr first.

Reply:
I am referring to `srtp_get_rtp_xtn_hdr_len()`, which always returns a value of 4 or more since it does not check the X bit. I think it should, because this creates an assumption that it was already checked.  While it is, I can imagine somebody changing the code and missing this fact.

If you want to save that `if` check, at least document the behavior with a comment.

I just prefer to check it explicitly.  I'd even be tempted to change the code calling this function to get the length and then act if the length is > 0.  But, I appreciate that causes yet another function call on every packet, though, unless the function is made inline.

Reply:
I get what you mean but the x bit is not part of srtp_hdr_xtnd_t which is what is passed to this function today, the assumtion is that if srtp_hdr_xtnd_t must be valid and in which case the length is always atleast 4. One could change the function to take a srtp_hdr_t and then do the x bit check but then it would need to find the srtp_hdr_xtnd_t after checking.
Will atleast document the assumtions, but let me know if you think I should change the input arrgument.